File: src/main/frontend/src/service/api/AwardService.js
import axios from "axios";
import {AWARD_API_URL} from "../constants";

export default class AwardService {

    static async getEmployeeAwards(employeeId) {
        return axios.get(`${AWARD_API_URL}/employee/${employeeId}`);
    }

    static async addAward(employeeId, award) {
        console.log(award)
        return axios.post(`${AWARD_API_URL}/employee/${employeeId}`, award)
    }
}

File: src/main/java/com/example/demo/service/AwardService.java
package com.example.demo.service;

import com.example.demo.rest.dto.award.AwardFull;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface AwardService {
    ResponseEntity<List<AwardFull>> getAwardsByEmployeeId(Long employeeId);

    void addAward(AwardFull request, Long employeeId);
}
